analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/HashesShouldHaveUnpredictableSaltBase.cs

```diff
@@ -68,20 +70,34 @@ private void ProcessObjectCreation(ProgramState state, IObjectCreationOperationW
         }
     }
 
-    private static ProgramState ProcessInvocation(ProgramState state, IInvocationOperationWrapper invocation) =>
-        IsCryptographicallyStrongRandomNumberGenerator(invocation)
-        && invocation.ArgumentValue("data") is { } dataArgument
-        && dataArgument.TrackedSymbol() is { } trackedSymbol
-            ? state.SetSymbolConstraint(trackedSymbol, ByteCollectionConstraint.CryptographicallyStrong)
-            : state;
+    private static ProgramState ProcessInvocation(ProgramState state, IInvocationOperationWrapper invocation)
+    {
+        if (IsCryptographicallyStrongRandomNumberGenerator(invocation)
+            && FirstMethodArgument(state, invocation) is { } dataArgument
```

Comment:
I'm afraid you need to incorporate the type check in this method (`FirstMethodArgument`) - otherwise, you get the first argument that might be an int and not the first argument that is a byte array.

In theory, this should not happen as these methods have one parameter - but the logic still has to be correct.